secantTests/LoadingScreenTests.swift

```diff
@@ -0,0 +1,239 @@
+//
+//  AppRouterNavigationTests.swift
+//  secantTests
+//
+//  Created by Francisco Gindre on 9/2/21.
+//
+
+import XCTest
+@testable import secant_testnet
+import Foundation
+import Combine
+class LoadingScreenTests: XCTestCase {
+    var cancellables: [AnyCancellable] = []
+    
+    // MARK: LoadingScreenViewModel Tests
+    
+    func testLoadingPublishedHomeScreenWhenCredentialsArePresent() throws {
+        let mockServices = MockServices()
```

Comment:
It looks like we go through alot of the same boilerplate setup code here.  Would it be good to factor this out into a helper that just takes the returnBlock as input so you don't have to write this same code for every test?

Reply:
Thanks for all the suggestions!

I don't think I'm quite following you here. Probably due to the code highlighting.

Reply:
It looks like every test runs though this code:
```
 let mockServices = MockServices()
         let stub = KeysPresentStub(returnBlock: {
             throw KeyStoringError.alreadyImported
         })
         mockServices.keyStorage = stub
         let loadingViewModel = LoadingScreenViewModel(services: mockServices)
          let testExpectation = XCTestExpectation(description: "LoadingViewModel Publishes .credentialsFound when credentials are present and there's no failure")
         let expected = LoadingScreenViewModel.LoadingResult.credentialsFound
 ```
I was just wondering if it would be worth it to move this to a setup function so you don't have to repeat yourself so much is all.  There are only very little changes between the different tests, but this is just something to consider so that as you add more tests you don't need to keep adding the same boilerplate in the future.

Reply:
The problem of extracting this is that the stub is lost and we would need to return a tuble with the stub with the spy boolean and the view model. I've tried it and I don't see much added value on the reusability.

Reply:
Sounds good!

Reply:
````
enum LoadingScreenViewModelHelper {
    static func loadingViewModelWith(keysPresentStubBlock: @escaping () throws -> Bool) -> LoadingScreenViewModel {
        let mockServices = MockServices()
        let stub = KeysPresentStub(returnBlock: keysPresentStubBlock)
        mockServices.keyStorage = stub
        return LoadingScreenViewModel(services: mockServices)
    }
}
````

This can be used when the Spy is not needed